consensus/src/sequencing_replica.rs

```diff
@@ -438,6 +462,19 @@ where
                     }
                     // starting from the first iteration with a three chain, e.g. right after the else if case nested in the if case above
                     if new_decide_reached {
+                        let mut leaf = leaf.clone();
+
+                        // If the full block is available for this leaf, include it in the leaf
+                        // chain that we send to the client.
+                        if let Some(block) =
+                            consensus.saved_blocks.get(&leaf.get_deltas_commitment())
+                        {
+                            if let Err(err) = leaf.fill_deltas(block.clone()) {
```

Comment:
If every node is a DA node, and we collect the async operations between views, the block should always be updated by the time this gets called. However, once we're doing small DA committee and distribution, this may often not yet be populated, and it would be better to have a way of updating the Option after Decide...

Reply:
I prefer having the client explicitly request the block using their commitment if it was not available at Decide, rather than having HotShot asynchronously populate it automatically, which would require some possibly messy shared state. But anyways, agreed this is something for the future